lasagna-master/gallen/lasagna_master.go

```diff
@@ -0,0 +1,31 @@
+package lasagna
+
+func PreparationTime(layers []string, avgPreparationTime int) int {
+    if (avgPreparationTime == 0) {
+        avgPreparationTime = 2
+    }
+    return len(layers) * avgPreparationTime
+}
+
+func Quantities(layers []string) (noodles int, sauce float64) {
+    for _, layer := range layers {
+        if (layer == "noodles") {
+            noodles += 50
+        } else if (layer == "sauce") {
+            sauce += 0.2
+        }
+    }
+    return
+}
+
+func AddSecretIngredient(friendsList, myList []string) []string {
+    return append(myList, friendsList[len(friendsList)-1])
+}
+
+func ScaleRecipe(amounts []float64, portions int) []float64 {
+    var result []float64
+    for _, v := range amounts {
+        result = append(result, v / float64(2) * float64(portions))
```

Comment:
You can extract `float64(portions) / 2.0` out of the loop.

Reply:
I don't understand.

Reply:
Only `v` changes within the loop but for each iteration you are doing `float64(portions) / 2.0` (or `1 / float64(2) * float64(portions)` in your code) which is a constant.

Reply:
I am not doing `float64(portions) / 2.0` anywhere in my loop.

Reply:
Yes, you are doing it here ` v / float64(2) * float64(portions`. That's literally `v` multiplied by `1 / float64(portions)` multiplied by `float64(portions)` being the last two a constant.

I explained myself like shit. Sorry for that :sweat_smile:

Reply:
Ah, now I get it. Thanks.

I think that would warrant a detailed comment to indicate what exactly is happening. While I now understand the math transformation you are making, it's not intuitive, at least to me.

Reply:
Absolutely right. I will next time!

Reply:
Change applied. Thanks for the feedback.